guides/data_modelling/cross_context_boundaries.md

```diff
@@ -0,0 +1,563 @@
+# 4. Cross-context Boundaries
+
+Now that we have the beginnings of our product catalog features, let's begin to work on the other main features of our application – carting products from the catalog. In order to properly track products that have been added to a user's cart, we'll need a new place to persist this information, along with point-in-time product information like the price at time of carting. This is necessary so we can detect product price changes in the future. We know what we need to build, but now we need to decide where the cart functionality lives in our application.
+
+If we take a step back and think about the isolation of our application, the exhibition of products in our catalog distinctly differs from the responsibilities of managing a user's cart. A product catalog shouldn't care about the rules of our shopping cart system, and vice-versa. There's a clear need here for a separate context to handle the new cart responsibilities. Let's call it `ShoppingCart`.
+
+Let's create a `ShoppingCart` context to handle basic cart duties. Before we write code, let's imagine we have the following feature requirements:
+
+  1. Add products to a user's cart from the product show page
+  2. Store point-in-time product price information at time of carting
+  3. Store and update quantities in cart
+  4. Calculate and display sum of cart prices
+
+From the description, it's clear we need a `Cart` resource for storing the user's cart, along with a `CartItem` to track products in the cart. With our plan set, let's get to work.
+
+## Adding authentication
+
+Most of the cart functionality is tied to a specific user. Therefore, in order to allow each user to manage their own cart (and only their own carts), we must be able to authentication users. To do so, we will use Phoenix' built-in `mix phx.gen.auth` generator to scaffold a solution for us:
```

Comment:
```suggestion
Most of the cart functionality is tied to a specific user. Therefore, in order to allow each user to manage their own cart (and only their own carts), we must be able to authenticate users. To do so, we will use Phoenix's built-in `mix phx.gen.auth` generator to scaffold a solution for us:
```

Oh and I asked AI if it is `Phoenix'` or `Phoenix's` and it said `Phoenix's`, which is what we use in the intro and other places.